lib/ast.ml

```diff
@@ -23,6 +23,8 @@ type t =
     | MFA of {module_name: t; function_name: t; arity: t}
     | ListCons of t * t
     | ListNil
+    | MapCreation of (t * t) list                  (* #{k1 => v1, ...} *)
+    | MapUpdate of t * (t * t) list * (t * t) list (* M#{k1 => v1, ..., ke1 := ve1, ...} *)
```

Comment:
How about using the following record?

```suggestion
    | MapUpdate of {map: t; assoc_fields: (t * t) list; exact_fields: (t * t) list} (* M#{k1 => v1, ..., ke1 := ve1, ...} *)
```

Reply:
Thanks! fixed: https://github.com/dwango/fialyzer/pull/141/commits/8061409a76017f97a424af7e5438b6aa637cc0c1